pkg/decode/decode_protobuf.go

```diff
@@ -26,6 +26,46 @@ const (
 type Protobuf struct {
 }
 
+// dnsRawNameToDotted parses a label-encoded DNS QNAME (raw bytes copied from kernel)
```

Comment:
this is a good candidate for unit-test as it looks so convoluted

Reply:
done in https://github.com/netobserv/netobserv-ebpf-agent/pull/820/commits/a4849cb1d206b1fea3ed73c72114f96f25d354b2